allennlp/training/metrics/fbeta_measure2.py

```diff
@@ -0,0 +1,172 @@
+        for example, to calculate a multi-class average ignoring a majority
+        negative class. Labels not present in the data will result in 0
+        components in a macro or weighted average.
+
+    index_to_label : `Dict[int, str]`, optional
+        A dictionary mapping indices to the corresponding label.
+        If this map is giving, the provided metrics include the label
+        instead of the index for each class.
+    """
+
+    def __init__(
+        self,
+        beta: float = 1.0,
+        average: Union[str, List[str]] = None,
+        labels: List[int] = None,
+        index_to_label: Dict[int, str] = None,
+    ) -> None:
+        super().__init__(beta=beta, average=None, labels=labels)
+        self._index_to_label = index_to_label
+        if not average:
+            average = ["micro"]
+        elif isinstance(average, str):
+            average = [average]
+
+        average_options = {"micro", "macro", "weighted"}
+        if not all([a in average_options for a in average]):
+            raise ConfigurationError(
+                f"`average` has to be `None` or one (or a list) of {average_options}."
+            )
+
+        self._averages = average
+
+    def get_metric(self, reset: bool = False):
+        """
+        # Returns
+
+        <class>-precision : `float`
+        <class>-recall : `float`
+        <class>-fscore : `float`
+        <avg>-precision : `float`
+        <avg>-recall : `float`
+        <avg>-fscore : `float`
+
+        where <class> is the index (or the label if `index_to_label` is given)
+        of each class or its label (in case `index_to_label` is given);
+        and <avg> is the option (or each one of the options) given in `average`.
+        """
+        if self._true_positive_sum is None:
+            raise RuntimeError("You never call this metric before.")
+
+        tp_sum = self._true_positive_sum
+        pred_sum = self._pred_sum
+        true_sum = self._true_sum
+
+        if self._labels is not None:
+            # Retain only selected labels and order them
+            tp_sum = tp_sum[self._labels]
+            pred_sum = pred_sum[self._labels]  # type: ignore
+            true_sum = true_sum[self._labels]  # type: ignore
+
+        beta2 = self._beta**2
+
+        # Finally, we have all our sufficient statistics.
+        precision = nan_safe_tensor_divide(tp_sum, pred_sum)
+        recall = nan_safe_tensor_divide(tp_sum, true_sum)
+        fscore = nan_safe_tensor_divide(
+            (1 + beta2) * precision * recall, beta2 * precision + recall
+        )
+
+        all_metrics = {}
+        for c, (p, r, f) in enumerate(zip(precision.tolist(), recall.tolist(), fscore.tolist())):
+            if self._index_to_label:
+                c = self._index_to_label[c]
+            all_metrics[f"{c}-precision"] = p
+            all_metrics[f"{c}-recall"] = r
+            all_metrics[f"{c}-fscore"] = f
+
+        if "macro" in self._averages:
```

Comment:
Since these calculations don't take any time, can you just do all of them every time? It doesn't hurt anything to have all the averages all the time, does it?

Reply:
Done. Renamed to FBetaVerboseMeasure and removed the average option (always return all three averages).